ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -722,12 +749,12 @@ public boolean isForeignKey() {
 
     @Override
     public List<? extends Expression> getOriginalViableKeys(final LogicalSchema schema) {
-      return leftSourceKeyColumns;
+      throw new UnsupportedOperationException("Should not be called with foreign key joins.");
```

Comment:
Good catch.

Reply:
Was actually pointed out by @vcrfxia on the original PR that added it (we just merge the other PR as-is, and I address open comments on this PR to close those gaps).